### ISSUE TITLE
Unable to change color of Dropdown parent button

### ISSUE THREAD

**Issue by meddouribadis** [themesberg/flowbite-react#911]
- [X] I have searched the [Issues](https://github.com/themesberg/flowbite-react/issues) to see if this bug has already been reported
- [X] I have tested the latest version

## Steps to reproduce

Below is a React function in which I pass tailwind CSS classes to my dropdown element, but the parent button does not receive the classes but it is the child elements of the dropdown that receive them instead. It's impossible to change the color of the Dropdown without creating a specific CSS class.

```js
function Dropdown({ name }) {
    return (
        <>
            <Dropdown className="bg-red-500 hover:bg-red-700 text-white font-bold py-2 px-4 rounded" label="Test">
            </Dropdown>
        </>
      )
}
```

## Current behavior

![Current behavior.](https://media.discordapp.net/attachments/922401356617244733/1139648616428998726/image.png)

Dropdown's parent button does not receive classes. The child elements yes.

## Expected behavior

The parent button should receive the classes and not the children. The way around this is to create a custom theme and activate the inline prop as in the example below which is a bit complicated to just change the color of the dropdown button.

```js
function Dropdown({ name}) {
    const customTheme = {
        "inlineWrapper": "text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center inline-flex items-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
    };

    return (
        <>
            <Dropdown inline={true} theme={customTheme} label="Test">
                <Dropdown.Header>
                    <span className="block text-sm">
                        Bonnie Green
                    </span>
                    <span className="block truncate text-sm font-medium">
                        bonnie@flowbite.com
                    </span>
                </Dropdown.Header>
            </Dropdown>
        </>
    )
}
```

## Context

I'm looking to add classes to my dropdown.

Thanks for your help !

**Comment by DavideFrancescon** [comment]
@meddouribadis @victorfunes  
Just checked the code for the component.
The dropdown button(it's actually a normal button with a custom function and a theme class added) has a theme key called target, changing that you should be able to change the button style. Let me know if this fixes things for you.

For adding classNames to the button instead of the items, you are right, I think it should be a default prop for the component.

**Comment by victorfunes** [comment]
Hi @DavideFrancescon ,
Maybe I am understanding something wrong. In my case, I wanted to change the width to full width, and following what I understood from your instructions, have tried changing the theme like this:

```
        <Dropdown
          label={selectedDateFilterText}
          color="light"
          className="w-full"
          theme={{ floating: { target: "w-full" } }}>
```

But it only affects to the floating menu and not to the button itself.
I have been looking the source code too, and I don't see where I can change the theme to change the button width.

Sorry if it is my mistake or missunderstanding.

**Comment by DavideFrancescon** [comment]
@victorfunes 
<img width="489" alt="Screenshot 2023-08-24 alle 17 11 30" src="https://github.com/themesberg/flowbite-react/assets/102860588/b40a920a-a597-4947-b0cb-4056ab7a5597">
<img width="1440" alt="Screenshot 2023-08-24 alle 17 12 40" src="https://github.com/themesberg/flowbite-react/assets/102860588/bcc35d97-3ba1-4fc1-b234-e501e8563c02">

<img width="656" alt="Screenshot 2023-08-24 alle 17 12 21" src="https://github.com/themesberg/flowbite-react/assets/102860588/386734d7-be0f-4dd5-9fb3-d2e5573b3d06">
<img width="1440" alt="Screenshot 2023-08-24 alle 17 12 10" src="https://github.com/themesberg/flowbite-react/assets/102860588/1b387c99-bb6d-4072-9803-02150a4ee4c7">

Using the theme object you sent seems to work, could you provide a little more of the page? 
What i'm thinking is that you are not giving the component enough space to notice the chances to the button, try checking for the parent div's width. The dropdown uses a separate "dimension" to render so it might not be inflicted by the parent's width.
(Tested the className and yes, that only applies to the dropdown container, so you can remove it, the theme applies to both of them).
Let me know if this fixes things, if not, maybe provide a sandbox with the full declarative code so I can help you find the issue!

**Comment by victorfunes** [comment]
Here it goes:

```
      <div className="block sm:hidden w-full">
        <Dropdown label={selectedDateFilterText} color="light" className="w-full">
          <Dropdown.Item onPointerDown={filterLastYear}>{t("filters.time.lastyear")}</Dropdown.Item>
          <Dropdown.Item onPointerDown={filterLastMonth}>{t("filters.time.last30days")}</Dropdown.Item>
          <Dropdown.Item onPointerDown={filterLastWeek}>{t("filters.time.last7days")}</Dropdown.Item>
          <Dropdown.Item onPointerDown={filterAlways}>{t("filters.time.always")}</Dropdown.Item>
        </Dropdown>
      </div>
```

![image](https://github.com/themesberg/flowbite-react/assets/108006690/03bf2528-6469-47d7-a1eb-bd3773a2f7c2)

**Comment by victorfunes** [comment]
My mistake I did not the last test with the theme, now it works:

      <div className="block sm:hidden w-full">
        <Dropdown label={selectedDateFilterText} color="light" theme={{ floating: { target: "w-full" } }}>
          <Dropdown.Item onPointerDown={filterLastYear}>{t("filters.time.lastyear")}</Dropdown.Item>
          <Dropdown.Item onPointerDown={filterLastMonth}>{t("filters.time.last30days")}</Dropdown.Item>
          <Dropdown.Item onPointerDown={filterLastWeek}>{t("filters.time.last7days")}</Dropdown.Item>
          <Dropdown.Item onPointerDown={filterAlways}>{t("filters.time.always")}</Dropdown.Item>
        </Dropdown>
      </div>

![image](https://github.com/themesberg/flowbite-react/assets/108006690/6fe8cd46-7f32-4ab7-a9a2-09eb96c424ef)

**Comment by DavideFrancescon** [comment]
@rluders @tulup-conner both this and the issue mentioned above can be closed I think.
But personally I would create a new one to update the docs to add a little tutorial on how to change every part of every component. So a schema of sort for every component where each part of it is linked to it's theme key.

**Comment by DavideFrancescon** [comment]
> @DavideFrancescon I agree with you, but I would recommend to change the theme name for the button in the Dropdown, as floating.target does not sound as the button style (it is not intuitive at all, as floating sounds like something related to the dropdown items menu, isn't it?)

Yes it's not, but every component has the same keys in the theme, so if I were to change this (other than break every old website using it), it would be the only one with a different name. So yea, it's not intuitive but it's too late to change it now.

**Comment by victorfunes** [comment]
I was refering not to change "target" but the the structure of the theme. 
Other components in Flowbite are using an structure representing the different subcomponents themes.
As an example I am using your Table component and the theme contains: root, body, head, row...
This to me was more intuitive.

**Comment by DavideFrancescon** [comment]
Yea it's a good solution, but it would still destroy every old site using the one there is now. The only solution is to leave both of them, and i'm in no position to do it without authorizations. 
@zoltanszogyenyi @tulup-conner  @rluders what do you think?

**Comment by tulup-conner** [comment]
Hey all, thanks for your patience. So I definitely agree that the `floating` name is confusing. The whole reason for that is that `<Dropdown>` is driven by [Floating UI](), which makes sense to me as the developer of the component, but I completely understand why that isn't intuitive to actual users now.

With that said, I can't really think of a better name. It is indeed the part of the `<Dropdown>` that effectively "floats". I'll keep this issue open as a discussion for suggestions to that end. Like @DavideFrancescon mentioned, it will break existing themes, but that's fine if we're able to find a more intuitive name - that's the point of running this all in a "alpha/beta" stage.
